doc/nrf/libraries/others/profiler.rst

```diff
@@ -13,129 +13,201 @@ The output is provided using RTT and can be visualized in a custom Python backen
+   .. code-block:: c
 
-	static void profile_data_event(uint32_t val1, int32_t val2, int16_t val3,
-				       uint8_t val4, const char *string)
-	{
-		struct log_event_buf buf;
+      static void profile_data_event(uint32_t val1, int32_t val2, int16_t val3,
+                  uint8_t val4, const char *string)
+      {
+        struct log_event_buf buf;
 
-		profiler_log_start(&buf);
-		/* Profiling data connected with an event */
-		profiler_log_encode_uint32(&buf, val1);
-		profiler_log_encode_int32(&buf, val2);
-		profiler_log_encode_int16(&buf, val3);
-		profiler_log_encode_uint8(&buf, val4);
-		profiler_log_encode_string(&buf, string);
-		profiler_log_send(&buf, data_event_id);
-	}
+        profiler_log_start(&buf);
+        /* Profiling data connected with an event */
+        profiler_log_encode_uint32(&buf, val1);
+        profiler_log_encode_int32(&buf, val2);
+        profiler_log_encode_int16(&buf, val3);
+        profiler_log_encode_uint8(&buf, val4);
+        profiler_log_encode_string(&buf, string);
+        profiler_log_send(&buf, data_event_id);
+      }
 
-.. note::
+   .. note::
 
-	The event ID and the data that is profiled with the event must be consistent with the registered event type.
-	The data for every data field must be provided in the correct order.
+	    The ``data_event_id`` and the data that is profiled with the event must be consistent with the registered event type.
+	    The data for every data field must be provided in the correct order.
 
+Configuration for use with Event Manager
+========================================
 
-Supported backends
-******************
+If you are using the Event Manager, complete the following steps to use the Profiler:
 
-The Profiler supports a custom backend to visualize the output data.
+1. Enable the Profiler by selecting the :kconfig:`CONFIG_EVENT_MANAGER_PROFILER` Kconfig option.
+   With this Kconfig option set, the Profiler is automatically initialized when you initialize the :ref:`event_manager`.
+#. Declare events for profiling purposes.
+   Read :ref:`event_manager_implementing_events` on the Event Manager documentation page to learn how to do this.
+#. Profile Event Manager events by following the steps in :ref:`event_manager_profiler_tracer_em_implementation` on the :ref:`event_manager_profiler_tracer` documentation page.
+
+.. _profiler_backends:
+
+Enabling supported backend
+**************************
+
+The Profiler supports a custom backend that is based around Python scripts to visualize the output data.
 The backend communicates with the host using RTT.
 
-Custom backend
-==============
+To save profiling data, the scripts use CSV files for event occurrences and JSON files for event descriptions.
 
-Select the custom backend to use dedicated tools written in Python for event visualization, analysis, and calculating statistics.
+The backend is available for both custom and Event Manager events.
```

Comment:
The device provides information about tracking Event Manager events using profiler events.

Reply:
I think that this sentence here is not necessary, I will remove it. The sentence you are proposing is already explained earlier.